### ISSUE TITLE
Any chance for Yi Dome 1080p support?

### ISSUE THREAD

**Issue by t00** [alienatedsec/yi-hack-v4#3]
@alienatedsec please consider building h20 image if possible, I will be happy to test. I tried using only home_y20 after renaming to home_h20, leaving original root_h20 from Keroi/yi-hack-v4 build - it did not work.

Can you possibly share some links to prepare own image - I am proficient with Linux, just going through the noise on Discord and having over 100 forks of v4 just overwhelms me to say at least.

Another question - should it be enough to just ssh contents of home partition to a running h20 image, replacing existing files? Maybe there is a way to mount home file using loop? I just need streaming support for home assistant with PTZ, ONVIF might do the trick.

**Comment by t00** [comment]
And one more question - why yi-hack-v4 should also be placed on SD card? Is the firmware reading contents from SD instead of home partition?

**Comment by alienatedsec** [comment]
Hi @t00 

Please use [this guide](https://github.com/alienatedsec/yi-hack-v4/wiki/Project-Setup-Hints-v2) to build your own firmware. It took me several attempts to be successful, but the guide is based on TheCrypt0 wiki + mine experience on Debian9.

The reason to offload the yi-hack-v4 from the internal memory to SD card is due to decompression problems and memory allocation which I have not resolved as of yet. Some cameras have also failed to boot if the home package of the hack version which includes RTSP, PTZ, ONVIF and other improvements. 

When you include newer versions of submodules, the size becomes a real issue; hence, offloading is my only option at the moment.

**Comment by alienatedsec** [comment]
@t00 
https://github.com/alienatedsec/yi-hack-v4/releases/tag/v0.2.3b

> @alienatedsec please consider building h20 image if possible, I will be happy to test. I tried using only home_y20 after renaming to home_h20, leaving original root_h20 from Keroi/yi-hack-v4 build - it did not work.

**Comment by Antarctic74** [comment]
@alienatedsec hi! I have a Yi dome 1080p (15us) and 2 Yi home 740p (47us) and they have already  yi hack 0.2.0 from TheCrypto. I see that the files h20.... and y18.... are already in your release so no need to download the TheCrypto files right? just format SD and copy your files?

**Comment by alienatedsec** [comment]
@Antarctic74 if you donated and obtained the license from TheCrypt0; you will lose it unless you have a backup (e.g. Email)

My release is not compatible with the v0.2.0 licensing model.

**Comment by Antarctic74** [comment]
@alienatedsec no I haven't paid for license thanks.
 I copied the files to the SD (formatted fat32), inserted the SD in the Yi dome 1080p and powered up the camera. But nothing happened, I connected to the web server and I still see v0.2.0 from TheCrypto... what's wrong? tried a restart but nothing.

**Comment by alienatedsec** [comment]
Wrong. All three part of the files have to be in the root folder.

**Comment by Antarctic74** [comment]
Sorry my mistake, unpacked the files for the 47us
Ok it works now, the Hostname can't be set, it's empty.
Now I'll try to configure it in my home assistant 
I'll report problems if any ;)

Great work @alienatedsec

**Comment by alienatedsec** [comment]
> Sorry my mistake, unpacked the files for the 47us
> 
> Ok it works now, the Hostname can't be set, it's empty.
> 
> Now I'll try to configure it in my home assistant 
> 
> I'll report problems if any ;)
> 
> 
> 
> Great work @alienatedsec 

When you restart or have the config, it will populate the hostname.

**Comment by t00** [comment]
@alienatedsec brillant, h20 firmware works great, only once streaming stopped until reboot when viewed both on yi app and VLC.
I have noticed a settings refresh delay being higher than @TheCrypt0 original h20 - rmm uses 70-80% CPU which I guess is normal for video processing?
Small issue was that symlinks are not supported on FAT32 and unpacking complained about it - not sure if any library symlink was needed - so far everything works including ONVIF integration in Home Assistant !
Either way I will give a go compiling it myself for fun - many thanks for a link, will use Debian 9 VM to avoid undiscovered issues :)

**Comment by Antarctic74** [comment]
@t00 please can you share RTSP / onvif / still image addresses?

**Comment by t00** [comment]
@Antarctic74 Sure - I use rtsp://IP/ch1_0.h264 to play video, there are other examples on https://github.com/roleoroleo/yi-hack-Allwinner-v2 All RTSP channels seem to be using same, pre-configured in web interface, high quality stream. I could not get snapshot to work though.
ONVIF auto-configuration in Home Assistant works out of the box - I just entered the IP address (not same subnet) and empty username/password. I guess default ports are in use.

**Comment by Antarctic74** [comment]
@t00 @alienatedsec for me the onvif port is 8080, no auth.
Low res stream is available at rtsp://192.168.1.140/ch0_1.h264 (profile_1 640*360) dunno why but this stream can't be viewed in Home assistant, OK in VLC)
Hi res stream is available at rtsp://192.168.1.140/ch0_0.h264  (profile_0 1920*1080) I can view this stream in Home assistant, with a lag of about 10 seconds. If I look the stream on VLC I have no lag, work great.

You can use this program to check onvif cameras [https://sourceforge.net/projects/onvifdm/](url)

**Comment by alienatedsec** [comment]
> 
> Low res stream is available at rtsp://192.168.1.140/ch0_1.h264 (profile_1 640*360) dunno why but this stream can't be viewed in Home assistant, OK in VLC)
> 
@Antarctic74 please check the known issues of the release. Also, it would be interesting to know if the stream is working stable through VLC.

**Comment by MiSSHeaDShoT** [comment]
Hello I will do also a test as I have Dome 1080p. Do you plan to update mqttv4 library to support baby crying and sound events? There is one here for others chipsets already made but I could not make it work for this firmware. I mean this release: 
https://github.com/roleoroleo/mqttv4

**Comment by alienatedsec** [comment]
> Hello I will do also a test as I have Dome 1080p. Do you plan to update mqttv4 library to support baby crying and sound events? There is one here for others chipsets already made but I could not make it work for this firmware. I mean this release:
> https://github.com/roleoroleo/mqttv4

@MiSSHeaDShoT lets hope for the next release integration :)

**Comment by Antarctic74** [comment]
Another issue is that the camera produce hires and lores streams even if you set to use only one.

**Comment by alienatedsec** [comment]
> Another issue is that the camera produce hires and lores streams even if you set to use only one.

@Antarctic74 - I hope to resolve it in the next release
![image](https://user-images.githubusercontent.com/73167064/113673190-94376180-96b0-11eb-882b-29f9510a3495.png)
